swagger.js

```diff
@@ -32,6 +32,53 @@ const doc = {
         },
       },
     },
+    definitions: {
+      Order: {
+        $id: '123456',
+        user: 'John Doe',
+        orderItems: [
+          {
+            album: '5f7f1f7a5c3f2b2d9c1b0b9d',
+            quantity: 2,
+          },
+        ],
+        subtotal: 9.99,
+        tax: 0.75,
+        total: 10.74,
+      },
+    },
```

Comment:
When I make a call to the GET /orders/{id} endpoint I get a response like this:

![](https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/9094098/2b792138-844c-4c66-a5b5-390db0fb4c62)

So we'll want to make sure our documented example response has the same shape:

```suggestion
      Order: {
        $_id: 'idabc123456',
        $user: {
          $id: 'idabc123456',
          $name: 'John Doe',
          $username: 'johndoe',
          $email: 'john@example.com',
          $role: { '@enum': ['user', 'admin'] }
        },
        $orderStatus: { '@enum': ['pending', 'payment_successful', 'payment_failed', 'cancelled', 'complete'], default: 'pending' },
        orderItems: [
          {
            $album: '5f7f1f7a5c3f2b2d9c1b0b9d',
            $quantity: 2,
            $_id: 'idabc12345'
          },
        ],
        $subtotal: 9.99,
        $tax: 0.75,
        $total: 10.74,
        $createdAt: '2023-08-08T20:45:00.942Z',
        $updatedAt: '2023-08-08T20:45:00.942Z',
        paymentIntentId: 'pi_abc123456'
      },
    },
```

The `$` in front of a field indicates that this is a required field or in the case of a response body, a field that is _always_ populated and returned.

The `@enum` field allows us to document that there is a finite set of options for this field (the `@` sign in front indicates that we're using raw Swagger/OpenAPI syntax here)

The resulting documentation should look like:

<img width="1017" alt="Screenshot 2023-09-02 at 10 46 47 PM" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/9094098/7b2e41f3-b704-4112-90a1-1ff8db214f99">